yorkshire/cli.py

```diff
@@ -49,9 +50,14 @@ def cli(debug: bool = False) -> None:
     type=str,
     metavar="FILE|URL|DIR",
 )
-def cli_detect(ctx: click.core.Context, path: str) -> None:
+@click.option(
+    '--allowed-index',
+    multiple=True,
+    help="allow this particular index",
+)
+def cli_detect(ctx: click.core.Context, path: str, allowed_indexes: Iterable[str]) -> None:
```

Comment:
Name of the argument needs to match the CLI option:

```suggestion
def cli_detect(ctx: click.core.Context, path: str, allowed_index_url: Tuple[str]) -> None:
```

Also, click passes a `Tuple` in this case.